### PR TITLE
GPG verification: conditionally select sigfiles + test suite

### PR DESCRIPTION
**Pull request by nathantypanski**
This is #24, #27, and #25. Plus a test suite so we can automatically verify that this can install the first and last version of Consul/Vault/Terraform from `asdf list-all`, as manual verification is tedious and error-prone.

Let me know what you think; I'm happy to make any modifications that will get this out the door.

### CODE REVIEW

**Comment by smorimoto** [comment]
@nathantypanski Anyway, I invited you to the @asdf-community/asdf-hashicorp team. If you are interested, feel free to take it.

**Comment by nathantypanski** [comment]
Gotta tweak the tests to pass in CI ...

File: test/install.bats
Comment (smorimoto) [inline test/install.bats:10]: I think it should be the repository root, not `.git` dir.

Comment (nathantypanski) [reply test/install.bats:8]: GitHub is running the workflow with https://github.com/asdf-community/asdf-hashicorp/pull/29/files#diff-126bf89616b7daa3d14ebc882ad18666aaf1c3dae888c4ba306a66ec80758bc1L52 I believe, and it's installing the Terraform plugin before these tests can even run.

Ideally, I'd test this against the _newer_  version of that file. Am I missing something?

Comment (nathantypanski) [reply test/install.bats:8]: I think if I move the plugin additions to there, this might "just work". Of course, it would require more local setup.

Fingers crossed.

**Comment by nathantypanski** [comment]
The tests now pass, but I want to hold off on merging for a second because I don't fully understand why. This _shouldn't_ work, because the `setup()` function in bats ought to be giving asdf a clean homedir. It must be running as root / some other user inside the test containers.

File: test/install.bats
Comment (smorimoto) [reply test/install.bats:8]: See this document, in this example, you are now specifying `/home/runner/work/my-repo-name/my-repo-name/.git`, not `/home/runner/work/my-repo-name/my-repo-name`.
https://docs.github.com/en/actions/reference/environment-variables

Comment (smorimoto) [reply test/install.bats:8]: Oops, disregard.

**Comment by smorimoto** [comment]
@nathantypanski I just pushed a few fixes.

**Comment by smorimoto** [comment]
Well, I think it's a good idea to set `ASDF_DIR` to temp for local testing.

**Review by smorimoto — APPROVED** [review]

**Comment by nathantypanski** [comment]
On my _local_ machine, this is blowing away my actual `asdf/plugins/vault` directory when I try to run  the tests. I'm trying to figure out something that will work both locally and in CI.

**Comment by nathantypanski** [comment]
@smorimoto this is now good to go. Can I get another approval for https://github.com/asdf-community/asdf-hashicorp/pull/29/commits/c58d5599ef232ddd96f9b2db1731af768cdd92ba so I'm not merging unreviewed commits?

File: test/install.bats
Comment (smorimoto) [inline test/install.bats:7]: Pass -t to mktemp to create a temporary directory under the /tmp directory.

Comment (nathantypanski) [reply test/install.bats:7]: Oh, yay, this is another OS incompatibility between Linux and BSD-derived OSes (e.g., macOS).

GNU `mktemp` deprecates `-t`, instead preferring `-p`:

```
       -p DIR, --tmpdir[=DIR]
              interpret TEMPLATE relative to DIR; if DIR is not  specified,
              use  $TMPDIR  if  set, else /tmp.  With this option, TEMPLATE
              must not be an absolute name; unlike with  -t,  TEMPLATE  may
              contain slashes, but mktemp creates only the final component

       -t     interpret  TEMPLATE as a single file name component, relative
              to a directory: $TMPDIR, if set; else the directory specified
              via -p; else /tmp [deprecated]
```

while BSD `mktemp` only supports `-t`.

Comment (smorimoto) [reply test/install.bats:7]: Hmm, okay let’s just go with -p :p

**Comment by smorimoto** [comment]
If CJ and Dustin OK, let's go with this. Thanks @nathantypanski 🙂

**Review by DustinChaloupka — APPROVED** [review]
Thanks!

**Pull request merged**
